### PR TITLE
docs: Add use case Environment Variables in Deployment Pipeline

### PR DESCRIPTION
**Pull request by alezander86**
Description
This documentation update introduces a detailed example of how to set test suite parameters using environment variables in Continuous Deployment pipelines. It focuses on leveraging the KubeRocketCI platform's capabilities for dynamic test suite configuration. This allows development teams to adjust their testing strategies without needing to alter the test suite code directly. The primary motivation for this change is to provide a more flexible, efficient, and quality-centric deployment pipeline that can adapt to various environments and project requirements.

Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement

Environment variables were successfully used to adjust test parameters dynamically in a deployment pipeline scenario.
The process was documented and verified for reproducibility and effectiveness in managing deployment pipeline configurations.

Checklist:
- [x] I have performed a self-review of my own documentation updates.
- [x] I have commented my updates, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation to reflect the new feature.
- [x] My changes generate no new warnings.
- [x] Pull Request contains one commit. I squash my commits.

### CODE REVIEW

File: docs/assets/use-cases/cd-autotests-run-with-env-variables/autotests-component-info.png
Comment (Rolika4) [inline docs/assets/use-cases/cd-autotests-run-with-env-variables/autotests-component-info.png:1]: add <repo_subpath>/autotest to git url path

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+To streamline the process of implementing custom logic in deployment flow quality gate within the KubeRocketCI platform, follow these steps:
+
+- Fork the Custom Tekton chart: Fork, clone and deploy the custom Tekton chart that implements the required deploy pipeline with the environment variables usage.
+- Add the Application: Add a simple application from the platform's marketplace and build it.
+- Add the Autotest: Add autotests prepared repository with the clone strategy.
+- Create the Deployment Flow, Environment, and Set Variables: Create a Deployment Flow, environment, add Autotests as a quality gate, define the necessary variables, and set them accordingly.
+- Validate the Implementation: Review the pipeline logs to ensure autotests execute correctly and produce the expected output.
+
+# Fork the Custom Tekton chart
```
Comment (Rolika4) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:27]: \###

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+5. Click on **BRANCHES** and click the **Trigger build PipelineRun** button:
+
+  ![Trigger build](../assets/use-cases/cd-autotests-run-with-env-variables/trigger-build.png "Trigger build")
+
+6. Wait until the build is successful:
+
+  ![Successful Build](../assets/use-cases/cd-autotests-run-with-env-variables/successful-build.png "Successful Build")
+
+# Add the Autotests
```
Comment (Rolika4) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:116]: \###

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+  * Start version from: `0.1.0`
+  * Suffix: `SNAPSHOT`
+
+  ![Autotests codebase advanced settings](../assets/use-cases/cd-autotests-run-with-env-variables/codebase-advanced-settings.png "Autotests codebase advanced settings")
+
+6. Click **Close**. Now the application and autotest are created.
+
+  ![Components overview](../assets/use-cases/cd-autotests-run-with-env-variables/components-overview.png "Components overview")
+
+# Create the Deployment Flow, Environment, and Set Variables
```
Comment (Rolika4) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:158]: \###

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+![Add application to deployment](../assets/use-cases/cd-autotests-run-with-env-variables/deployment-add-application.png "Add application to deployment")
+
+6. Click the **CREATE ENVIRONMENT** button, define the following values, and click **NEXT**:
+
+  * Cluster: `In cluster`
+  * Stage name: `qa`
+  * Description: `qa`
+  * Trigger type: `Manual`
+  * Pipeline template: `deploy-with-autotests-variables`
+
```
Comment (Rolika4) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:189]: Clean pipeline template: clean

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+- **Add the Application**: Add a simple application from the platform marketplace and build the application to prepare it for deployment.
+- **Add the Autotests**: Set up the autotests with the clone strategy to include them in the Deployment Flow.
+- **Create Deployment Flow, Environment, and Set Variables**: Create a Deployment Flow, Set up an environment for the deployment, add Autotests as a quality gate to the flow, and define the variable.
+- **Validate**: Review the pipeline logs to ensure autotests execute correctly and produce the expected output.
+
+### Fork the Custom Tekton Chart
+
+The first step is to clone and install the custom Tekton chart:
+
+1. Create a private repository and copy the [chart repository](https://github.com/alezander86/custom-tekton-chart) content. Below is the repository structure:
```
Comment (MykolaMarusenko) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:31]: why we use alezander86 account?

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+3. Update the component configuration in the `values.yaml` file:
+
+    ```yaml
+    global:
+      # Replace `example.domain.com` with the appropriate wildcard domain for your setup.
+      dnsWildCard: "example.domain.com"
+    ```
+
+4. Update Helm dependencies:
```
Comment (MykolaMarusenko) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:76]: do we need this dependency? in upstream we do not use this approach 
https://github.com/epmd-edp/helm-helm-pipeline/blob/master/Chart.yaml

File: docs/use-cases/cd-autotests-run-with-env-variables.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+  ![Autotests component](../assets/use-cases/cd-autotests-run-with-env-variables/codebase-autotest.png "Autotests component")
+
+3. Choose **Clone project** strategy and click the **CREATE** button:
+
+  ![Autotests clone strategy](../assets/use-cases/cd-autotests-run-with-env-variables/codebase-autotests-clone.png "Autotests clone strategy")
+
+4. Fill in the following fields in **Add components info** and click the **NEXT** button to proceed:
+
+  * RepoUrl: `https://github.com/SergK/autotests`
```
Comment (MykolaMarusenko) [inline docs/use-cases/cd-autotests-run-with-env-variables.md:136]: maybe we can migrate into krci or epmd-edp account?

**Review by MykolaMarusenko — APPROVED** [review]

**Pull request merged**